apps/portal/src/components/operators/OperatorFilters.tsx

```diff
@@ -1,17 +1,101 @@
-import React from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
+import { Search, X } from 'lucide-react';
+import { Input } from '@/components/ui/input';
+import { useOperatorFilters } from '@/hooks/use-operators';
 
 interface OperatorFiltersProps {
   loading?: boolean;
 }
 
-export const OperatorFilters: React.FC<OperatorFiltersProps> = ({ loading = false }) => (
-  <div className="space-y-4">
-    {/* Loading indicator only */}
-    {loading && (
-      <div className="flex items-center justify-end space-x-2 text-sm text-muted-foreground">
-        <div className="w-4 h-4 border-2 border-primary border-t-transparent rounded-full animate-spin" />
-        <span>Loading...</span>
+export const OperatorFilters: React.FC<OperatorFiltersProps> = ({ loading = false }) => {
+  const { filters, updateSearch, toggleMyStakesOnly } = useOperatorFilters();
+  const [localSearch, setLocalSearch] = useState(filters.searchQuery);
+
+  // Sync local state when filters are reset externally
+  useEffect(() => {
+    setLocalSearch(filters.searchQuery);
+  }, [filters.searchQuery]);
+
+  // Debounced search update
+  useEffect(() => {
+    const timeoutId = setTimeout(() => {
+      if (localSearch !== filters.searchQuery) {
+        updateSearch(localSearch);
+      }
+    }, 300);
```

Comment:
Can we extract this into constant to give better context. Something like `const SEARCH_DEBOUNCE_MS = 300;`